File: src/main/java/com/naminhyeok/fantazzk/room/application/GetRoom.java
package com.naminhyeok.fantazzk.room.application;

import com.naminhyeok.fantazzk.CoreException;
import com.naminhyeok.fantazzk.room.domain.Room;
import com.naminhyeok.fantazzk.room.domain.RoomErrorType;
import com.naminhyeok.fantazzk.room.repository.Rooms;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class GetRoom {
    private final Rooms rooms;

    @Transactional(readOnly = true)
    public Room get(String code) {
        return rooms.findByCode(code).orElseThrow(() -> CoreException.of(RoomErrorType.ROOM_NOT_FOUND));
    }
}

File: src/main/kotlin/com/naminhyeok/fantazzk/room/application/GetRoom.kt
package com.naminhyeok.fantazzk.room.application

import com.naminhyeok.fantazzk.room.domain.Room
import com.naminhyeok.fantazzk.room.exception.RoomException
import com.naminhyeok.fantazzk.room.repository.Rooms
import org.springframework.stereotype.Component

@Component
class GetRoom(
    private val roomRepository: Rooms,
) {
    fun get(code: String): Room = roomRepository.findByCode(code) ?: throw RoomException.RoomNotFoundException()
}
